### PR TITLE
Optimize List.chop

### PR DESCRIPTION
**Pull request by vasily-kirichenko**

### CODE REVIEW

**Comment by cartermp** [comment]
Link #5084

File: src/absil/illib.fs [hunk 1]
```diff
@@ -280,7 +280,6 @@ module List =
         | h::t -> if f h then Some (h,n) else findi (n+1) f t
 
     let chop n l = 
-        if n = List.length l then (l,[]) else // avoids allocation unless necessary
```
Comment (smoothdeveloper) [inline src/absil/illib.fs]: List.length is deadly, I wonder if some of the places shouldn't use array, in particular the things that we tend to look at list length from.

**Comment by cartermp** [comment]
@vasily-kirichenko Do you have a scenario under profile where you can show the improvement? Logically this change makes sense to me (`List.length` feels like it should never be used like this).

**Comment by dsyme** [comment]
For micro-optimization my guess is that we should just replace with `List.splitAt`

**Comment by vasily-kirichenko** [comment]
Closing in favor of https://github.com/Microsoft/visualfsharp/pull/5090

**Comment by dsyme** [comment]
@vasily-kirichenko Is it possible a GC triggered in the middle of `List.chop` and the profiler accounted that to that function?

**Comment by vasily-kirichenko** [comment]
@dsyme no, GC waits are shown in dotTrace explicitly.

**Pull request closed without merging**